AutomatedTesting/Gem/PythonTests/Physics/tests/tick/Tick_CharacterGameplayComponentMotionIsSmooth.py

```diff
@@ -0,0 +1,97 @@
+    character_motion_smooth    = ("Character motion passed smoothness threshold", "Failed to meet smoothness threshold for character motion")
+# fmt: on
+
+
+def Tick_CharacterGameplayComponentMotionIsSmooth():
+    """
+    Summary:
+     Create entity with PhysX Character Controller and PhysX Character Gameplay components.
+     Verify that the motion of the character controller under gravity is smooth.
+
+    Expected Behavior:
+     1) The motion of the character controller under gravity is a smooth curve, rather than an erratic/jittery movement.
+
+    Test Steps:
+     1) Load the empty level
+     2) Create an entity
+     3) Add a PhysX Character Controller Component and PhysX Character Gameplay component
+     4) Enter game mode and collect data for the character controller's z co-ordinate and the time values for a series of frames
+     5) Check if the motion of the character controller was sufficiently smooth
+
+    :return: None
+    """
+    # imports
+    import os
+    import azlmbr.legacy.general as general
+    import azlmbr.math as math
+    from editor_python_test_tools.editor_entity_utils import EditorEntity as Entity
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+    from editor_python_test_tools.asset_utils import Asset
+    import numpy as np
+
+    # constants
+    COEFFICIENT_OF_DETERMINATION_THRESHOLD = 1 - 1e-4 # curves with values below this are not considered sufficiently smooth
+
+    helper.init_idle()
+    # 1) Load the empty level
+    helper.open_level("Physics", "Base")
+
+    # 2) Create an entity
+    test_entity = Entity.create_editor_entity("test_entity")
+    Report.result(Tests.create_entity, test_entity.id.IsValid())
+
+    azlmbr.components.TransformBus(
+        azlmbr.bus.Event, "SetWorldTranslation", test_entity.id, math.Vector3(0.0, 0.0, 0.0))
+
+    # 3) Add character controller and character gameplay components
+    character_controller_component = test_entity.add_component("PhysX Character Controller")
+    Report.result(Tests.character_controller_added, test_entity.has_component("PhysX Character Controller"))
+    character_gameplay_component = test_entity.add_component("PhysX Character Gameplay")
+    Report.result(Tests.character_gameplay_added, test_entity.has_component("PhysX Character Gameplay"))
+
+    # 4) Enter game mode and collect data for the rigid body's z co-ordinate and the time values for a series of frames
+    t = []
+    z = []
+    general.enter_game_mode()
+    general.idle_wait_frames(1)
+    game_entity_id = general.find_game_entity("test_entity")
+    for timestep in range(100):
+        t.append(azlmbr.components.TickRequestBus(azlmbr.bus.Broadcast, "GetTimeAtCurrentTick").GetSeconds())
+        z.append(azlmbr.components.TransformBus(azlmbr.bus.Event, "GetWorldZ", game_entity_id))
+        general.idle_wait_frames(1)
```

Comment:
Suggestion Instead of single ticks you could do 100 ticks at once and have a callback on the tickbus

Reply:
Good to know, although I quite like the simplicity of this approach.